cli/dcoscli/config/main.py

```diff
@@ -237,19 +252,31 @@ def _unset(name, index):
         if index is not None:
             index = util.parse_int(index)
 
+            if not value:
+                raise DCOSException(
+                    'Index ({}) is out of bounds - {} is empty'.format(
+                        index,
+                        name))
             if index < 0 or index >= len(value):
                 raise DCOSException(
                     'Index ({}) is out of bounds - possible values are '
                     'between {} and {}'.format(index, 0, len(value) - 1))
 
-            value.pop(index)
+            popped_value = value.pop(index)
+            emitter.publish(
+                "[{}]: removed element '{}' at index '{}'".format(
```

Comment:
The colon after the property name here is inconsistent with output from other commands. 

E.g.:
```
➜  cli git:(DCOS-2379-show-config-changes) ✗ dcos config set core.timeout 5
[core.timeout] changed from '3' to '5'
```

And:
```
➜  cli git:(DCOS-2379-show-config-changes) ✗ dcos config unset --index=1 package.sources
[package.sources]: removed element 'https://github.com/mesosphere/universe/archive/version-1.y.zip' at index '1'
```

Is the name at the front a strict requirement (which isn't a bad idea)? If so, add a colon to the other messages. Alternatively, change this to `"Removed element '{}' at index '{}' from [{}]"`